multithreading/future.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <memory>
+#include <functional>
+
+namespace accelerated {
```

Comment:
Rather change the namespace `recorder` (avoids conflicts if `accelerated-arrays` is also inlcuded in the project)